### PR TITLE
Python3 support, Scrapy 1.0+ dependencies

### PR DESCRIPTION
**Pull request by nyov**
Since f34749322e005b546318f30d5fd7d3dc329c932a, because `scrapy.loader` does not exist prior.

### CODE REVIEW

**Comment by nyov** [comment]
Well that is awkward. I wasn't planning on fixing tests.

**Comment by nyov** [comment]
Rebased on top of #71
Added some things to *try* fix bsddb3 bytes keys. Test succeeds now but something looks strange (values converted to bytes on return).

File: tests/test_deltafetch.py [hunk 1]
```diff
@@ -86,8 +87,8 @@ def test_spider_opened_existing(self):
         mw.spider_opened(self.spider)
         assert hasattr(mw, 'db')
         assert isinstance(mw.db, type(dbmodule.db.DB()))
-        assert mw.db.items() == [('test_key_1', 'test_v_1'),
-                                 ('test_key_2', 'test_v_2')]
+        assert mw.db.items() == [(b'test_key_1', b'test_v_1'),
+                                 (b'test_key_2', b'test_v_2')]
```
Comment (nyov) [inline tests/test_deltafetch.py]: This is strange: `test_v_1` values are returned as bytes. But were inserted as strings (`db.put(b'test_key_1', 'test_v_1')`). Not sure if this is bsddb3 not casting back to int/string on fetch or what.

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
+six
 boto
 hubstorage
```
Comment (redapple) [inline requirements.txt]: let's require `hubstorage>=0.23`

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
 python-dateutil
 scrapinghub
-Scrapy>=0.22.0
+Scrapy>=1.1.0rc1
```
Comment (redapple) [inline requirements.txt]: Scrapy 1.1 being out, let's require `scrapy>=1.1`

**Comment by redapple** [comment]
@nyov , I re-triggered the builds on Travis.
All green now

File: setup.py [hunk 4]
```diff
@@ -16,5 +16,5 @@
         'Operating System :: OS Independent',
         'Programming Language :: Python'
     ],
-    install_requires=['Scrapy>=0.22.0']
+    install_requires=['Scrapy>=1.0.0']
```
Comment (redapple) [inline setup.py]: Should it require same scrapy version as in `requirements.txt`?

File: setup.py [hunk 5]
```diff
@@ -16,5 +16,5 @@
         'Operating System :: OS Independent',
         'Programming Language :: Python'
```
Comment (redapple) [inline setup.py]: What about declaring Python 2 and Python 3 compatibility?

File: setup.py (hunk 4 above)
Comment (redapple) [inline setup.py]: ... maybe not, so as to support Scrapy 1.0 users

Comment (nyov) [inline setup.py]: I'm not sure. Perhaps the line can be dropped completely?

File: setup.py (hunk 5 above)
Comment (nyov) [inline setup.py]: I would think `Programming Language :: Python` is generic already.
But if there is good reason to explicitly define `Programming Language :: Python :: 2` and `Programming Language :: Python :: 3` (or even minor versions?) I have no problem adding them.

**Comment by nyov** [comment]
Updated and rebased on master.
(This includes/outdates the https://github.com/scrapinghub/scrapylib/tree/py3-porting branch)

**Comment by nyov** [comment]
So I guess this commit won't make it...?

I've just been staring at a new deployment error 
```py
  File "/usr/local/lib/python3.5/dist-packages/scrapylib/processors/__init__.py", line 5, in <module>
    from urlparse import urljoin
ImportError: No module named 'urlparse'
```
And I had this deja-vu (didn't I fix this already?)...

It's really great to see the new packages coming, but I would think a transitional package version would be great as well. Pull in those new depends and strip the package piecemeal.
(Or where did those `scrapylib.processors` end up?)

**Comment by redapple** [comment]
@nyov , you're right, a new release of scrapylib with new depends makes sense.
The migration effort to scrapy-plugins has stalled right now and `scrapylib.processors` is still only in scrapylib

**Comment by nyov** [comment]
Thanks for the info, @redapple.

Should `processors` simply be moved to scrapy proper, perhaps?
If it ever comes to that¹ , I think it might make sense to package up ItemLoaders and their processors together into a package.
Creating a package for these few processors, but keeping the rest in scrapy doesn't seem sensible.

(¹ IIRC ItemLoader was kept from contrib rather than packaged itself, after several(?) discussions.)

**Comment by redapple** [comment]
Thanks a ton @nyov .
The switch to scrapy-plugins continues in https://github.com/scrapinghub/scrapylib/pull/76

**Pull request merged**